components/common/src/main/java/com/hotels/styx/config/schema/Schema.java

```diff
@@ -217,12 +325,42 @@ public FieldValue elementType() {
         public FieldValue elementType() {
             return elementType;
         }
+
+        @Override
+        public Optional<String> isCorrectType(List<String> parents, JsonNode parent, JsonNode element, Function<String, FieldValue> schemas) {
+            return element.isObject() ? validateMap(parents, parent, element, schemas) : message(parents, element);
+        }
+
+        @Override
+        public String describe() {
+            return format("MAP (%s)", elementType.describe());
+        }
+
+        private Optional<String> validateMap(List<String> parents, JsonNode parent, JsonNode mapNode, Function<String, FieldValue> schemas) {
+            AtomicReference<Optional<String>> success = new AtomicReference<>(Optional.empty());
+
+            mapNode.fieldNames().forEachRemaining(key -> {
+                JsonNode entry = mapNode.get(key);
+                Optional<String> correctType = elementType.isCorrectType(push(parents, format("%s", key)), mapNode, entry, schemas);
+
+                if (correctType.isPresent()) {
+                    success.set(correctType);
+                }
+            });
+            return success.get();
+        }
+
+        private Optional<String> message(List<String> parents, JsonNode tree) {
+            String fullName = Joiner.on(".").join(parents);
+            return Optional.of(format("Unexpected field type. Field '%s' should be MAP ('%s'), but it is %s", fullName, this.elementType().describe(), tree.getNodeType()));
```

Comment:
*Here why I decided to add the comment to remove hardcoded strings for the field type. Still a minor issue of course...

Reply:
The element type can be anything. Even a nested map. For this reason the `describe()` is really important. If I understood your comment correctly? :-)